### PR TITLE
Add error modals for error states for JS desktop application

### PR DESCRIPTION
**Pull request by matthugs**
https://openedx.atlassian.net/browse/EDUCATOR-3689

### CODE REVIEW

File: edx_proctoring/static/proctoring/js/exam_action_handler.js [hunk 1]
```diff
@@ -24,6 +24,31 @@ var edx = edx || {};
...
   };
 
+  /**
+   * Launch modals, handling a11y focus behavior
+   *
+   * Note: don't try to leverage this for the heartbeat; the DOM
+   * structure this depends on doesn't live everywhere that handler
+   * needs to live
+   */
+   function accessibleError(message) {
```
Comment (matthugs) [inline edx_proctoring/static/proctoring/js/exam_action_handler.js]: The error modal this generates will be visually consistent with the modal shown on the instructions page when folks try to start the exam without having finished the desktop application setup.

File: edx_proctoring/templates/proctored_exam/ready_to_start.html [hunk 2]
```diff
@@ -45,14 +45,39 @@ <h3>
...
-          Start my exam
-        {% endblocktrans %}
+      {% trans "Start my exam" as start_exam %}
+      <button type="button" class="exam-action-button proctored-enter-exam btn btn-pl-primary btn-base" data-action="start" data-exam-id="{{exam_id}}" data-change-state-url="{{change_state_url}}" data-loading-text="<span class='fa fa-circle-o-notch fa-spin'></span> Starting Exam" data-cta-text="{{ start_exam }}">
+        {{ start_exam }}
       </button>
     </div>
   </div>
 </div>
+<div id="accessible-error-modal" class="modal" aria-hidden="true">
```
Comment (matthugs) [inline edx_proctoring/templates/proctored_exam/ready_to_start.html]: this html also nabbed from the instructions page

File: edx_proctoring/templates/proctored_exam/ready_to_submit.html [hunk 3]
```diff
@@ -28,6 +27,32 @@ <h3>
     </button>
   {% endif %}
 </div>
+<div id="accessible-error-modal" class="modal" aria-hidden="true">
```
Comment (davestgermain) [inline edx_proctoring/templates/proctored_exam/ready_to_submit.html]: How about moving these modals to the footer.html, defining it only once and filling in the error message dynamically?

File: edx_proctoring/templates/proctored_exam/ready_to_submit.html [hunk 4]
```diff
@@ -28,6 +27,32 @@ <h3>
...
+  <div class="inner-wrapper" role="dialog" aria-labelledby="accessible-error-title">
+    <button class="close-modal">
+      <span class="icon fa fa-remove" aria-hidden="true"></span>
+      <span class="sr">
+          {% trans "Close" %}
+      </span>
+    </button>
+
+    <header>
+        <h2 id="acessible-error-title">{% trans "Cannot Start Proctored Exam" %}</h2>
```
Comment (davestgermain) [inline edx_proctoring/templates/proctored_exam/ready_to_submit.html]: This should be `Cannot Stop Proctored Exam`

File: edx_proctoring/templates/proctored_exam/ready_to_submit.html (hunk 3 above)
Comment (matthugs) [reply edx_proctoring/templates/proctored_exam/ready_to_submit.html]: Surprising detail: ready_to_submit doesn't use that footer. Looking at the commit history I'm not confident that discrepancy is unintentional; there's copy in the footer with which we may not want to distract the user.

Comment (davestgermain) [reply edx_proctoring/templates/proctored_exam/ready_to_submit.html]: Now this same modal is in three templates. Surely there's a way to refactor this into one place.

Comment (matthugs) [reply edx_proctoring/templates/proctored_exam/ready_to_submit.html]: please don't call me shirley

**Pull request merged**